modules/auxiliary/admin/sap/sap_igs_xmlchart_xxe.rb

```diff
@@ -146,7 +150,7 @@ def get_file_content(html_response)
     @file_content = file_content.gsub('>', '')
   end
 
-  def analyze_first_response(html_response)
+  def analyze_first_response(html_response, check = false)
```

Comment:
I'm unclear on why this option was added as it appears that it was added to check if this method was being called from the `check` method, yet I would expect that you would always want to print out the line `System is vulnerable, but the file #{@file} was not found on the host #{@host}` regardless of where it was called from?

This comment also applies to the rest of the lines you changed below.

Reply:
This option has been added because in case of check we read the file /etc/os-release, but if it is not in the system, then this function will display: `System is vulnerable, but the file /etc/passwd was not found on the host 10.10.10.10`

Reply:
Hmm ok, in this case why not just change the code so that we pass in the file name into the `analyze_first_response` call? This would allow you to still print out the fact that the file doesn't exist, which is important info for users to know, whilst also being able to handle the case where the `@file` variable may not match the file that we are currently trying to retrieve. It should also help simplify the code somewhat.

Reply:
https://github.com/rapid7/metasploit-framework/pull/14163/commits/d230599efd45eeaa8e2a36bd96a25df4b23fca5c

Reply:
Module output after change:
```
msf6 auxiliary(admin/sap/sap_igs_xmlchart_xxe) > check

[*] System is vulnerable, but the file /etc/os-release was not found on the host 172.16.30.29
[+] 172.16.30.29:40080 - The target is vulnerable.
msf6 auxiliary(admin/sap/sap_igs_xmlchart_xxe) > run
[*] Running module against 172.16.30.29

[+] File: /etc/passwd saved in: /Users/vladimir/.msf4/loot/20201002192604_SAP_TEST_172.16.30.29_sap.igs.xxe_065679.txt
[*] Auxiliary module execution completed
```

Reply:
Awesome thanks for making this change 👍 Marking this as resolved